File: android/app/src/main/java/com/ideals/arnav/route/RouteModels.kt
package com.ideals.arnav.route

data class LatLng(
    val lat: Double,
    val lng: Double
)

data class TurnStep(
    val maneuverType: String,       // "turn", "depart", "arrive", "continue", etc.
    val maneuverModifier: String?,  // "left", "right", "slight left", "uturn", etc.
    val instruction: String,        // "Turn left onto Main St"
    val distance: Double,           // meters for this step
    val duration: Double,           // seconds for this step
    val location: LatLng            // where maneuver occurs
)

data class RouteInfo(
    val waypoints: List<LatLng>,
    val distanceMeters: Double,
    val durationSeconds: Double,
    val steps: List<TurnStep> = emptyList()
)

/** Result of snapping user position to route. */
data class SnapResult(
    val snappedLat: Double,
    val snappedLng: Double,
    val segmentIndex: Int,
    val distanceToRoute: Float,
    val distanceAlongRoute: Float
)

File: ios/ARNavigation/ARNavigation/Route/RouteModels.swift
import Foundation
import CoreLocation

struct LatLng: Codable, Equatable {
    let lat: Double
    let lng: Double

    var coordinate: CLLocationCoordinate2D {
        CLLocationCoordinate2D(latitude: lat, longitude: lng)
    }
}

struct RouteInfo {
    let waypoints: [LatLng]
    let distanceMeters: Double
    let durationSeconds: Double
}

/// Result of snapping user position to route.
struct SnapResult {
    let snappedLat: Double
    let snappedLng: Double
    let segmentIndex: Int
    let distanceToRoute: Float
    let distanceAlongRoute: Float
}
